### DISCUSSION TITLE
react 18 stream ssr

### DISCUSSION THREAD

**Discussion by MrWangJustToDo** [emotion-js/emotion#2788] [Q&A] [answered]
when i use emotion and react 18 to create a stream ssr temaplte, i got a weird problem.
when page hydrate, i got some duplicate <style /> element which have same content, I'm not sure this is expected behavior or not
![Screenshot from 2022-06-20 22-34-38](https://user-images.githubusercontent.com/57748455/174625294-a45311a2-71a3-45ee-bd6f-0bd88256c9dc.png)

I'm using chakra ui which base on the emotion, so any one can give some help, thanks 
sorry i can not give a live sanbox to show this, you can visit [ssr-setup](https://github.com/MrWangJustToDo/react-ssr-setup)

i also create a discussion on chakra repo, see [https://github.com/chakra-ui/chakra-ui/discussions/6207](https://github.com/chakra-ui/chakra-ui/discussions/6207)

**Comment by MrWangJustToDo** [comment]
so sad

**Comment by Andarist** [comment, marked as answer]
You can wrap your root with a Chakra-aware component, or just with our CacheProvider. It should help you to deduplicate those styles.

**Reply by MrWangJustToDo** [reply]
oh thanks, it's work. i think it will work just like next.js, but look like there are some different
